document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
```

Comment:
No, we should change this, the requirements around the readable and non-readable ids are very clear based on background with PREfast and FxCop and they don't apply to tool errors. In that context, 'opaque' here certainly does mean that the id doesn't communicate information to a human. A rule requires a stable identifier for a couple of clear reasons: 1) so that specific checks can be enabled/disabled in scripts, etc., 2) to refer to in baselines to track ongoing review. If you change a rule id, you risk breaking lab infrastructure or reraising issues to users that they have seen before. Because an id needs to be extremely stable, it shouldn't try to convey any meaningful information to users that could change over time. A rule 'DoNotFooOrBar' might have a change in implementation at some point, because a security issue in 'Bar' was resolved. Meanwhile, the 'DoNotFooOrBar' guidance can persist in baselines, in-source suppressions, etc., forever conveying outdated or inaccurate guidance. 

A tool error is a different beast. Tools should not provide the ability to selectively disable them or to ignore tool errors in baselines: any notification with a severity of error should fail, period. Tool developers are free to change these ids at any time or to convey whatever information makes sense. Think of all this like the result of GetHashCode implementation. Users should not take a dependency on a specific GetHashCode result forever expressing something about the associated object. That's a bug. That id is specific/internal to the class. Same thing with a tool error id.